source/cal.c

```diff
@@ -33,7 +33,10 @@ static struct aws_error_info s_errors[] = {
     AWS_DEFINE_ERROR_INFO_CAL(
         AWS_ERROR_CAL_UNSUPPORTED_ALGORITHM,
         "The specified algorithim is unsupported on this platform."),
-};
+    AWS_DEFINE_ERROR_INFO_CAL(
```

Comment:
should we add error type for tag/aad not matching in GCM?

Reply:
I don't know how to detect a bad tag is the reason the encryption operations failed, and IIRC cryptographers don't tell you that due to various tampering attack vectors being able to use that information to craft a better attack.

Reply:
According to docs if EVP_DecryptFinal fails in AEAD mode then it can be assumed authn part failed and im guessing mac/win would have something similar. Havent worked with GCM at c level, but i know that java/ts exposed bad tag as exception. But if thats no longer a recommended practice, im fine with hiding it. 

relevant part from openssl docs:
When decrypting, the return value of EVP_DecryptFinal() or EVP_CipherFinal() indicates whether the operation was successful. If it does not indicate success, the authentication operation has failed and any output data MUST NOT be used as it is corrupted.
https://www.openssl.org/docs/man3.0/man3/EVP_DecryptFinal_ex.html